internal/privatemessaging/privatemessaging.go

```diff
@@ -298,11 +298,7 @@ func (pm *privateMessaging) sendData(ctx context.Context, tw *fftypes.TransportW
 				batch.Namespace,
 				batch.Payload.TX.ID,
 				fftypes.OpTypeDataExchangeSendBatch)
-			var groupHash *fftypes.Bytes32
-			if tw.Group != nil {
```

Comment:
The `tw.Group` object is only set on un-pinned transfers, where we can't guarantee the group has already been confirmed in an earlier pinned transfer. This code meant we weren't persisting the group for retry for pinned batches.

![image](https://user-images.githubusercontent.com/6660217/162096094-4599f06e-930b-4cd8-a0b1-00b13b2dc112.png)